### PR TITLE
feat(conf): use hocon schema

### PR DESCRIPTION
**Pull request by z8674558**
Fixes <issue-number>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

File: src/emqx_schema.erl [hunk 1]
```diff
@@ -0,0 +1,1443 @@
...
+        , {"strict_mode", boolean("emqx.strict_mode", false)}
+        , {"response_information", string("emqx.response_information", undefined)}
+    ];
+
+fields("zone") ->
+    [ {"$name", ref("zone_settings")}];
+
+fields("zone_settings") ->
+    [ {"idle_timeout", duration(undefined, "15s")}
+        , {"allow_anonymous", fun boolean/1}
```
Comment (k32) [inline src/emqx_schema.erl:233]: Nit: indentation.

File: bin/emqx [hunk 2]
```diff
@@ -123,8 +123,8 @@ fi
 # Echo to stderr on errors
 echoerr() { echo "$@" 1>&2; }
 
-# By default, use cuttlefish to generate app.config and vm.args
-CUTTLEFISH="${USE_CUTTLEFISH:-yes}"
+# By default, use hocon to generate app.config and vm.args
```
Comment (zmstone) [inline bin/emqx:126]: I don't think anyone is actually using hand-crafted app.conf and vm.args any more.
i.e. we can drop this support in 5.0

File: bin/emqx [hunk 3]
```diff
@@ -303,7 +304,7 @@ if [ -z "$NAME_ARG" ]; then
             NODENAME="$(grep -E '^-name' "$LATEST_VM_ARGS" | awk '{print $2}')"
         else
             # for boot commands, inspect emqx.conf for node name
-            NODENAME=$("$ERTS_PATH"/escript "$RUNNER_ROOT_DIR"/bin/cuttlefish -i "$REL_DIR"/emqx.schema -c "$RUNNER_ETC_DIR"/emqx.conf get node.name)
+            NODENAME=$(grep -E '^[ \t]*node.name[ \t]*=[ \t]*' "$RUNNER_ETC_DIR/emqx.conf" 2> /dev/null | tail -1 | awk -F"= " '{print $NF}'| tr -d \")
```
Comment (zmstone) [inline bin/emqx:307]: why are we going back to the old approach ?
the issue it was: environment variables were respected.

Comment (z8674558) [reply bin/emqx:307]: I will add `get` command to hocon escript soon

File: bin/emqx (hunk 2 above)
Comment (zmstone) [reply bin/emqx:126]: I'll help to remote it.

**Comment by z8674558** [comment]
check also https://github.com/emqx/emqx-ct-helpers/pull/48

**Review by zmstone — APPROVED** [review]

**Pull request merged**